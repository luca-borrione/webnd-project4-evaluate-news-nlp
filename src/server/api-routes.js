const express = require('express');
const { postAnalyse, getTest } = require('./api-routes-controller');

const router = express.Router();

router.route('/analyse').post(postAnalyse);
router.route('/test').get(getTest);

module.exports = router;
