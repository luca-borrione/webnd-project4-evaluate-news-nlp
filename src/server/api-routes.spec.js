describe('api-routes', () => {
  let routeSpy;
  let routeGetSpy;
  let routePostSpy;

  function prepareRouteSpies(path) {
    const express = require('express'); // eslint-disable-line global-require

    routeGetSpy = jest.fn();
    routePostSpy = jest.fn();
    routeSpy = jest.fn().mockImplementation((routePath) => {
      if (routePath === path) {
        return {
          get: routeGetSpy,
          post: routePostSpy,
        };
      }
      return {
        get: jest.fn(),
        post: jest.fn(),
      };
    });
    jest.spyOn(express, 'Router').mockReturnValueOnce({
      route: routeSpy,
    });
    require('./api-routes'); // eslint-disable-line global-require
  }

  beforeEach(() => {
    jest.resetModules();
  });

  it('should match the routes stored in the snapshot', () => {
    prepareRouteSpies();
    expect(routeSpy.mock.calls).toMatchSnapshot();
  });

  it('should correctly set the route for /analyse', () => {
    const route = '/analyse';
    prepareRouteSpies(route);
    const routeIndex = routeSpy.mock.calls.findIndex((mockCall) => mockCall[0] === route);
    expect(routeIndex).toBeGreaterThanOrEqual(0);
    const callback = routePostSpy.mock.calls[0][0];
    expect(callback.name).toBe('postAnalyse');
  });

  it('should correctly set the route for /test', () => {
    const route = '/test';
    prepareRouteSpies(route);
    const routeIndex = routeSpy.mock.calls.findIndex((mockCall) => mockCall[0] === route);
    expect(routeIndex).toBeGreaterThanOrEqual(0);
    const callback = routeGetSpy.mock.calls[0][0];
    expect(callback.name).toBe('getTest');
  });
});
