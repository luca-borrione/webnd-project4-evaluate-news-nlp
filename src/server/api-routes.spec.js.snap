// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api-routes should match the routes stored in the snapshot 1`] = `
Array [
  Array [
    "/analyse",
  ],
  Array [
    "/test",
  ],
]
`;
